src/action-ros-ci.ts

```diff
@@ -630,7 +630,9 @@ done`;
 
 	if (isLinux) {
 		// Always update APT before installing packages on Ubuntu
-		await execShellCommand(["sudo apt-get update"]);
+		await execShellCommand([
+			"apt update || sudo apt update || dnf update || sudo dnf update",
```

Comment:
I'd prefer doing having Ubuntu- and RHEL-specific commands here instead of running them all and hoping one works.

Reply:
That means explicitly checking for the Linux distribution and then checking against a database of distributions and their package manager. This database has to be kept up-to-date for all possible Debian and RHEL forks, and those distributions that use e.g. `dnf` but are not forks for RHEL. That sounds error-prone to me. Is this really desired?

What do you think about using PackageKit, i.e. `pkcon update`, to update the package index instead of distribution-specific commands?

Reply:
I think you could just do the same `if` you did in https://github.com/ros-tooling/setup-ros/pull/694.

`setup-ros` and `action-ros-ci` do not really intend to support Debian-based distros that aren't Ubuntu, so for RHEL you could keep it simple too.

However, to make it more flexible, for Ubuntu, you could even use `ID_LIKE` from `/etc/os-release`, which is `debian` on Ubuntu and (I assume) other Debian-based distros. Not sure what `ID_LIKE` looks like for RHEL-related distros.

Reply:
> What do you think about using PackageKit, i.e. pkcon update, to update the package index instead of distribution-specific commands?

Feel free to give it a shot if you want, but you'd need to install it first, so not sure it's worth the download just to avoid an `if`.

Reply:
> I think you could just do the same `if` you did in [ros-tooling/setup-ros#694](https://github.com/ros-tooling/setup-ros/pull/694).

I see you already discovered the first half of my mini project of porting the ROS GitHub actions to RHEL ? RPM.

> Feel free to give it a shot if you want, but you'd need to install it first, so not sure it's worth the download just to avoid an `if`.

You are right; ' pkcon` would require installing PackageKit first, defeating the purpose of not dealing with the native package managers.

In the end, I added a new function `determineDistrib` to extract the distribution name and did the if-else call of the package managers.